pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -360,6 +360,16 @@ func validateAWS(m *Machine, clusterID string) (bool, utilerrors.Aggregate) {
 		)
 	}
 
+	if providerSpec.Placement.Region == "" {
```

Comment:
Exposing the region to the API was chosen in a moment where there was more unknowns and under different assumptions than we can make today.
Given the current context where the region is implicit in the cluster there's other paths we could consider:
1 - This is not required. The backend infers this info from the infrastructure object.
2 - In addition to this change we default to what the infrastructure object.

Thoughts?

Reply:
Since we have the field now, I would suggest that defaulting it from the infrastructure object seems like a sensible approach, would mean we don't have to make changes to the behaviour of the controller, which could break people (eg if they're trying to create machines in a different region?)

Reply:
>Since we have the field now, I would suggest that defaulting it from the infrastructure object seems like a sensible approach, 

I'm ok with that.

>which could break people (eg if they're trying to create machines in a different region?)

That'd be impossible as a cluster is tied to specific region by design, right?
So there's only one valid value for this field which is the one where the cluster exists. So this being a user choice is bad UX as it just gives the users chances to shoot themselves on the foot.

Reply:
> That'd be impossible as a cluster is tied to a specific region by design, right?
So there's only one valid value for this field which is the one where the cluster exists. So this being a user choice is bad UX as it just gives the users chances to shoot themselves on the foot.

Yes I agree it would be shooting people in the foot, we could use webhooks to ensure it's the right value but I guess that might break someone who might have tried to create machines in another region or something? 

If you think we should just ignore the region field we can do, but that would be a breaking change right?

Reply:
I think anyone trying that is already broken by design. But I agree defaulting on webhook seems sensible.